content/en/blog/2020/large-scale-security-policy-performance-tests/index.md

```diff
@@ -0,0 +1,158 @@
+
+{{< image width="90%" ratio="45.34%"
+    link="AuthZ_paths_vs_sourceIP.svg"
+    alt="Authorization policy with both paths and `sourceIP`"
+    caption=""
+    >}}
+The latency increase of a variable number of `sourceIP` rules is marginally greater than that of path rules.
+{{< /tab >}}
+
+{{< tab name="AuthZ paths" category-value="five" >}}
+{{< image width="90%" ratio="45.34%"
+    link="AuthZ_var_paths.svg"
+    alt="Authorization policy with variable number of paths"
+    caption=""
+    >}}
+The latency increase of a single `AuthZ` policy with 10 path rules is not proportional to the latency increase of a single `AuthZ` policy with 1000 path rules compared to the system with sidecar and no policies. This trend is similar to that of `sourceIP` rules.
+{{< image width="90%" ratio="45.34%"
+    link="AuthZ_paths_vs_sourceIP.svg"
+    alt="Authorization policy with both paths and `sourceIP`"
+    caption=""
+    >}}
+The latency of a variable number of paths rules is marginally lesser than that of `sourceIP` rules.
+{{< /tab >}}
+
+{{< tab name="RequeatAuthN JWT Issuer" category-value="six" >}}
+{{< image width="90%" ratio="45.34%"
+    link="RequestAuthN_jwks.svg"
+    alt="Request Authentication with variable number of JWT issuers"
+    caption=""
+    >}}
+The latency of a single JWT issuer is comparable to that of no policies, but as the number of JWT issuers increase, the latency increases disproportionately.
+{{< /tab >}}
+
+{{< tab name="Variable AuthZ" category-value="seven" >}}
+To test how the number of Authorization policies affect runtime, the tests can be broken into two cases:
+
+- 1: Every Authorization policy has a single `sourceIP` rule.
+
+- 2: Every Authorization policy has a single path rule.
+
+{{< image width="90%" ratio="45.34%"
+    link="AuthZ_var_policies_sourceIP.svg"
+    alt="Authorization policy with variable number of policies, with `sourceIP` rule"
+    caption=""
+    >}}
+{{< image width="90%" ratio="45.34%"
+    link="AuthZ_var_policies_paths.svg"
+    alt="Authorization policy with variable number of policies, with path rule"
+    caption=""
+    >}}
+The overall trends of both graphs are similar. This is consistent to the paths vs `sourceIP` data, which showed that the latency is marginally greater for `sourceIP` rules than that of path rules.
+{{< /tab >}}
+
+{{< /tabset >}}
+
+## Conclusion
+
+- In general, adding security policies does not add relatively high overhead to the system. The policies that add the most latency include:
+
+    1. Authorization policy with `JWTRules` rules.
+
+    1. Authorization policy with `requestPrincipal` rules.
+
+    1. Authorization policy with principals rules.
+
+- In lower loads (requests with lower qps and conn) the difference in latency for most policies is minimal.
+
+- Envoy proxy sidecars increase latency more than most policies, even if the policies are large.
```

Comment:
I don't fully understand this.  What does this means to users?  not worry about policies because envoy itself increase latency more even when istio policies aren't used?

Reply:
We wanted to let the users know that the increase of latency due to security policies is not that great, that the security policies are overall quite minimal in relation to the increase of latency